### PR TITLE
[docs] PR for Update hardware requirement with SSE 4.2

### PR DESCRIPTION
**Pull request by lizayugabyte**
Also edited the doc for clarity

### CODE REVIEW

File: docs/content/preview/deploy/checklist.md [hunk 1]
```diff
@@ -61,79 +59,83 @@ You should allocate adequate CPU and RAM. YugabyteDB has good defaults for runni
...
 
 YugabyteDB stores data compressed by default. The effectiveness of compression depends on the data set. For example, if the data has already been compressed, then the additional compression at the storage layer of YugabyteDB will not be very effective.
 
-Plan for about 20% headroom on each node to allow space for miscellaneous overheads such as temporary additional space needed for compactions, metadata overheads, etc.
+It is recommended to plan for about 20% headroom on each node to allow space for miscellaneous overheads such as temporary additional space needed for compactions, metadata overheads, and so on.
 
 ### Network
 
-Below is a minimal list of default ports (along with the network access required) in order to use YugabyteDB.
+The following is a minimal list of default ports (along with the network access required) required to use YugabyteDB:
```
Comment (ddhodge) [inline docs/content/preview/deploy/checklist.md:115]:
```suggestion
The following is a minimal list of default ports (along with the network access) required to use YugabyteDB:
```

File: docs/content/preview/deploy/checklist.md [hunk 2]
```diff
@@ -61,79 +59,83 @@ You should allocate adequate CPU and RAM. YugabyteDB has good defaults for runni
...
+Memory depends on your application query pattern. Writes require memory but only up to a certain point (for example, 4GB, but if you have a write-heavy workload you may need a little more). Beyond that, more memory generally helps improve the read throughput and latencies by caching data in the internal cache. If you do not have enough memory to fit the read working set, then you will typically experience higher read latencies because data has to be read from disk. Having a faster disk could help in some of these cases.
 
-YugabyteDB explicitly manages a block cache, and doesn't need the entire data set to fit in memory. It doesn't rely on the OS to keep data in its buffers. If you give YugabyteDB sufficient memory, data accessed and present in block cache stays in memory.
+YugabyteDB explicitly manages a block cache, and does not need the entire data set to fit in memory. It does not rely on the OS to keep data in its buffers. If you provide YugabyteDB sufficient memory, data accessed and present in block cache stays in memory.
 
 ### Verify support for SSE2
```
Comment (ddhodge) [inline docs/content/preview/deploy/checklist.md:68]: and SSE4.2?

**Review by ddhodge — APPROVED** [review]
LGTM

File: docs/content/preview/deploy/checklist.md (hunk 2 above)
Comment (lizayugabyte) [reply docs/content/preview/deploy/checklist.md:68]: Yes! thank you

File: docs/content/preview/deploy/checklist.md (hunk 1 above)
Comment (lizayugabyte) [reply docs/content/preview/deploy/checklist.md:115]: Fixed

**Pull request merged**